syntax = "proto3";

package chat;

message Note {
    string sender = 1;
    string recipient = 2;
    string message = 3;
}

message Response {
  string message = 1;
  int32 status = 2;
  int32 sender = 3;
}

message String {
  string message = 1;
}

message Empty {}

// The greeting service definition.
service Greeter {
  // Sends a greeting
//  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc ValidateUser (UserName) returns (UserValidation) {}
  rpc ChatStream (UserName) returns (stream Note);
  rpc SendNote (Note) returns (Response);
  rpc ListAccounts (String) returns (Response);
  rpc DeleteAccount (UserName) returns (Response);
  rpc CreateAccount (UserName) returns (Response);
  rpc Login (UserName) returns (Response);
  rpc Disconnect (UserName) returns (Response);

}

// The request message containing the user's name.
message UserName {
  string name = 1;
}

// The response message containing the greetings
message UserValidation {
  bool message = 1;
}

